src/Symfony/Component/Workflow/EventListener/AuditTrailListener.php

```diff
@@ -0,0 +1,55 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Workflow\EventListener;
+
+use Psr\Log\LoggerInterface;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\Workflow\Event\Event;
+
+/**
+ * @author Grégoire Pineau <lyrixx@lyrixx.info>
+ */
+class AuditTrailListener implements EventSubscriberInterface
+{
+    public function __construct(LoggerInterface $logger)
+    {
+        $this->logger = $logger;
+    }
+
+    public function onLeave(Event $event)
+    {
+        foreach ($event->getTransition()->getFroms() as $place) {
+            $this->logger->info(sprintf('leaving "%s" for subject of class "%s"', $place, get_class($event->getSubject())));
+        }
+    }
+
+    public function onTransition(Event $event)
+    {
+        $this->logger->info(sprintf('transition "%s" for subject of class "%s"', $event->getTransition()->getName(), get_class($event->getSubject())));
+    }
+
+    public function onEnter(Event $event)
+    {
+        foreach ($event->getTransition()->getTos() as $place) {
+            $this->logger->info(sprintf('entering "%s" for subject of class "%s"', $place, get_class($event->getSubject())));
+        }
+    }
+
+    public static function getSubscribedEvents()
+    {
+        return array(
+            'workflow.leave' => array('onLeave'),
```

Comment:
I saw that after yeah...
Actually I don't think it will be that much usage for a such class as in most case the general event is too general (except for logging which is already implemented...)